### PR TITLE
plugins/persisted: init

### LINKED ISSUE #2603: [PLUGIN REQUEST] persisted.nvim
| Field    | Description |
|----------|-------------|
| Plugin   | `persisted.nvim`  | 
| Homepage | `https://github.com/olimorris/persisted.nvim`     | 
| Nixpkgs  | `pkgs.vimPlugins.persisted-nvim` | 

## Extra Information

Thanks and good luck

### PR DESCRIPTION
**Pull request by GaetanLepage**
Add support for [persisted.nvim](https://github.com/olimorris/persisted.nvim), a plugin for simple session management.

Fixes #2603

### CODE REVIEW

File: plugins/by-name/persisted/default.nix [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        ''
+          Function to determine if a session should be saved
+
+          ```lua
+          @type fun(): boolean
+          ```
+        '';
+
+    save_dir =
+      defaultNullOpts.mkStr { __raw = "vim.fn.expand(vim.fn.stdpath('data') .. '/sessions/')"; }
```
Comment (MattSturgeon) [inline plugins/by-name/persisted/default.nix:37]: We can use the new `literalLua` for doc strings like this (https://github.com/nix-community/nixvim/pull/2520)

**Review by MattSturgeon — COMMENTED** [review]
I don't see any real issues

File: plugins/by-name/persisted/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/by-name/persisted/default.nix]: Sure, done.

File: plugins/by-name/persisted/default.nix [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      Whether to include the git branch in the session file name.
+    '';
+
+    autoload = defaultNullOpts.mkBool false ''
+      Whether to automatically load the session for the cwd on Neovim startup.
+    '';
+
+    on_autoload_no_session = defaultNullOpts.mkRaw "function() end" ''
+
+      Function to run when `autoload = true` but there is no session to load
```
Comment (khaneliman) [inline plugins/by-name/persisted/default.nix:54]:
```suggestion
      Function to run when `autoload = true` but there is no session to load.
```

File: plugins/by-name/persisted/default.nix [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+      icons =
+        defaultNullOpts.mkAttrsOf types.str
+          {
+            selected = " ";
+            dir = "  ";
+            branch = " ";
+          }
+          ''
+            Icons displayed in the Telescope picker
```
Comment (khaneliman) [inline plugins/by-name/persisted/default.nix:89]:
```suggestion
            Icons displayed in the Telescope picker.
```

File: plugins/by-name/persisted/default.nix [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    allowed_dirs = defaultNullOpts.mkListOf types.str [ ] ''
+      List of dirs that the plugin will start and autoload from.
+    '';
+
+    ignored_dirs = defaultNullOpts.mkListOf types.str [ ] ''
+      List of dirs that are ignored for starting and autoloading.
+    '';
+
+    telescope = {
```
Comment (khaneliman) [inline plugins/by-name/persisted/default.nix:69]: If they have a telescope integration, it's nice to create an option to add it to list of extensions to load. See https://github.com/nix-community/nixvim/blob/e2f81c8e8e8baa28b100e0e43b721f16de6299d8/plugins/by-name/project-nvim/default.nix#L118-L127

Comment (khaneliman) [reply plugins/by-name/persisted/default.nix:69]: Looks like it would be `persisted` 
> require("telescope").load_extension("persisted")

Comment (GaetanLepage) [reply plugins/by-name/persisted/default.nix:82]: Good idea ! Done.

**Review by khaneliman — APPROVED** [review]
LGTM

**Comment by GaetanLepage** [comment]
@mergifyio queue

**Pull request merged**